generate_compile_commands.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+echo "Generate compile_commands.json with settings $@"
+make $@ --always-make --dry-run \
+ | grep -wE 'gcc|g\+\+' \
+ | grep -w '\-c' \
+ | jq -nR '[inputs|{directory:".", command:., file: match(" [^ ]+$").string[1:]}]' \
```

Comment:
'jq' command should be installed before building the project. So could you add the ‘jq’ installation command to the Makefile or the generate_compile_commands.sh file？

Reply:
The `jq` application should be a prerequisite of the building environment together with `git` and `arm-none-eabi-*`, so I think it should go into a Readme file that specifies the building environment. 
On the other hand, the implement of auto-installation will bind the Makefile to a particular kind of Linux release. For example, using `apt install` in the Makefile limits the building script onto `Debain`-based OS. 
Thus, I'll add a readme file into this pull request and document the prerequisite of this project into it. Let me know if it makes sense.